### DISCUSSION TITLE
Unable to Launch Vendure Project on Railway with Admin UI Extensions

### DISCUSSION THREAD

**Discussion by MaartenWessels** [vendurehq/vendure#2875] [Q&A] [answered]
Hello Vendure Community,

I am facing an issue while trying to launch my Vendure project in a testing environment on Railway. Here are the steps I followed:

1. Created a new Vendure installation using the command provided in the documentation.
2. Did the setup for Railway with the official docs: [deploy-to-railway](https://docs.vendure.io/guides/deployment/deploy-to-railway/)
3. Installed `@vendure/ui-devkit` using `npm install @vendure/ui-devkit`.
4. Updated my `vendure-config` with the following code:

```javascript
AdminUiPlugin.init({
    route: "admin",
    port: 3002,
    app: compileUiExtensions({
        outputPath: path.join(__dirname, '../admin-ui'),
        extensions: [{
            translations: {
                nl: path.join(__dirname, '../src/translations/nl.json'),
            }
        }],
    }),
    adminUiConfig:{
        defaultLanguage: LanguageCode.nl,
        availableLanguages: [LanguageCode.nl, LanguageCode.nl],
    }
}),
```

5. Added the translation file inside the translation folder.
6. Ran `npm install` and `npm run build` successfully on Railway.
7. Started the application using `node ./dist/index.js`.

However, when I check the logs, I get the following error:

```
/bin/sh: 1: ng: not found
error Command failed with exit code 127.
```

Despite the application running, I am unable to open the admin interface or use the api's

**Additional Information:**
- Vendure version: 2.2.4
- Node.js version: v20.12.2
- Railway setup details: [deploy-to-railway](https://docs.vendure.io/guides/deployment/deploy-to-railway/)

Could anyone please help me resolve this issue? I am unsure why the `ng` command is not found even though all dependencies are installed. Any guidance or suggestions would be greatly appreciated.

Thank you!

**Comment by michaelbromley** [comment, marked as answer]
I'm guessing that Railway might have the `NODE_ENV` env var set to `production` which would then prevent devDependencies from getting installed. 

In general, I'd recommend you look into [compiling the admin ui as a deploy step](https://docs.vendure.io/guides/extending-the-admin-ui/getting-started/#compiling-as-a-deployment-step). The reason is that with your current configuration, the custom admin ui app will get compiled _every time the Vendure server is started_. Not only does it use quite a lot of RAM (about 2GB), but it is also pretty slow. Best to do it once up-front.

**Reply by MaartenWessels** [reply]
Yes, I believe you're right. Railway automatically used the Dockerfile instead of my custom command, and compiling the admin UI in the build process works much better. Thank you for your quick help! I'm finding that Vendure is very developer-friendly, even for a front-end dev like me just starting out.

**Reply by michaelbromley** [reply]
Great to hear! You're also welcome to join our Discord (https://vendure.io/community) where you can find lots of community members who can help & answer your questions.
